### ISSUE TITLE
Don't push when in conflict

### ISSUE THREAD

**Issue by folofjc** [michal-h21/vimwiki-sync#13]
I have had a couple instances where there is a conflict on more than one file. So I open a file in vim, fix the conflict, and then close back to the cli. But then vimwiki-sync does a commit and push, which removes _other_ files from conflict state by adding them, and then commits them even though they have the conflict markers in them. 

Can you make it where it doesn't do anything if the branch is in conflict state?

**Comment by michal-h21** [comment]
Good point. Do you know what command could be used to detect a conflict?

**Comment by folofjc** [comment]
If this command returns anything, then there is a conflict:
`git diff --name-only --diff-filter=U`

Also, `git diff --check` looks for conflict markers, so you could also see if that is empty

**Comment by michal-h21** [comment]
Sorry, it took me some time, but I've just added the check for Git conflicts before committing. I hope it fixes this issue.

**Comment by folofjc** [comment]
No problem, thanks for this! It really is mainly an issue when you first start putting this on multiple computers. After that, it takes care of itself, haha!

**Comment by michal-h21** [comment]
Yes, I've never had this issue until today, when the `tags` file somehow managed to end up in Git, even though I have it listed in the `.gititgnore` file.

**Comment by folofjc** [comment]
Hmmmm. I think if you do a `git add` on a file, even if it is in the `.gitignore` file, it will still get added. The `.gitignore` just doesn't show you the files when you do `git status`. But you can always override it by adding the file explicitly.
